crates/cheatcodes/src/crypto.rs

```diff
@@ -230,30 +241,33 @@ fn sign_with_wallet(
 }
 
 fn sign_p256(private_key: &U256, digest: &B256) -> Result {
-    ensure!(*private_key != U256::ZERO, "private key cannot be 0");
-    let n = U256::from_limbs(*p256::NistP256::ORDER.as_words());
-    ensure!(
-        *private_key < n,
-        format!("private key must be less than the secp256r1 curve order ({})", n),
-    );
-    let bytes = private_key.to_be_bytes();
-    let signing_key = P256SigningKey::from_bytes((&bytes).into())?;
+    let signing_key = parse_private_key_p256(private_key)?;
     let signature: Signature = signing_key.sign_prehash(digest.as_slice())?;
     let r_bytes: [u8; 32] = signature.r().to_bytes().into();
     let s_bytes: [u8; 32] = signature.s().to_bytes().into();
 
     Ok((r_bytes, s_bytes).abi_encode())
 }
 
-fn parse_private_key(private_key: &U256) -> Result<SigningKey> {
+fn validate_private_key<C: ecdsa::PrimeCurve>(private_key: &U256) -> Result<()> {
```

Comment:
we could pass either C by value to use Debug or a &str parameter for the name of the curve, to use in the error message